cache/lru_sized_cache_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package cache
+
+import (
+	"testing"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSizedLRU(t *testing.T) {
+	cache := NewSizedLRU[ids.ID, TestSizedInt](TestSizedIntSize)
+
+	TestBasic(t, cache)
+}
+
+func TestSizedtLRUEviction(t *testing.T) {
+	cache := NewSizedLRU[ids.ID, TestSizedInt](2 * TestSizedIntSize)
+
+	TestEviction(t, cache)
+}
+
+func TestSizedLRUResize(t *testing.T) {
+	require := require.New(t)
+	cacheIntf := NewSizedLRU[ids.ID, TestSizedInt](2 * TestSizedIntSize)
+	cache, ok := cacheIntf.(*sizedLRU[ids.ID, TestSizedInt])
+	require.True(ok)
+
+	id1 := ids.ID{1}
+	id2 := ids.ID{2}
+
+	expectedVal1 := TestSizedInt{i: 1}
+	expectedVal2 := TestSizedInt{i: 2}
+	cache.Put(id1, expectedVal1)
+	cache.Put(id2, expectedVal2)
+
+	val, found := cache.Get(id1)
+	require.True(found)
+	require.Equal(expectedVal1, val)
+
+	val, found = cache.Get(id2)
+	require.True(found)
+	require.Equal(expectedVal2, val)
+
+	cache.maxSize = TestSizedIntSize
```

Comment:
This tests mimicks what happen with lru cache and currently fails given that we have chosen to implement resize differently. Not sure if we want to keep same behaviour for lru and sized lru cache

Reply:
We don't